src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -72,16 +81,14 @@ subsumes' val (TypeApp f1 r1) (TypeApp f2 r2) | f1 == tyRecord && f2 == tyRecord
                    -- What happens next is a bit of a hack.
                    -- TODO: in the new type checker, object properties will probably be restricted to being monotypes
                    -- in which case, this branch of the subsumes function should not even be necessary.
-                   case r2' of
-                     REmpty -> throwError . errorMessage $ AdditionalProperty p1
-                     _ -> unifyTypes r2' (RCons p1 ty1 rest)
+                   unifyTypes r2' (RCons p1 ty1 rest)
                    go ts1 ((p2, ty2) : ts2) r1' rest
     | otherwise = do rest <- freshType
-                     case r1' of
-                       REmpty -> throwError . errorMessage $ PropertyIsMissing p2
-                       _ -> unifyTypes r1' (RCons p2 ty2 rest)
+                     unifyTypes r1' (RCons p2 ty2 rest)
                      go ((p1, ty1) : ts1) ts2 rest r2'
+  firstMissingProp t2 t1 = fst <$> (uncons $ minusBy' (comparing fst) t2 t1)
```

Comment:
Could you please add a comment here too? Sorry to be so picky, but I'm trying to improve documentation in the typechecker as I go these days.